src/context/MapContext.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Dispatch,
+  MutableRefObject,
+  ReactNode,
+  SetStateAction,
+  createContext,
+  useRef,
+  useState,
+} from 'react';
+
+import { Extent } from 'ol/extent';
+import BaseLayer from 'ol/layer/Base';
+import Layer from 'ol/layer/Layer';
+import Map from 'ol/Map';
+import { useDebounce } from 'react-use';
+
+import { useFilters } from '@/hooks/useFilters';
+import { getStacCollections } from '@/services/stac';
+import { Collection } from '@/typings/stac';
+
+export type MapContextType = {
+  mapConfig: MapConfig;
+  setMapConfig: Dispatch<SetStateAction<MapConfig>>;
+  mapRef: MutableRefObject<HTMLDivElement | null>;
+  map: Map | undefined;
+  setMap: Dispatch<SetStateAction<Map | undefined>>;
+  getLayers: () => BaseLayer[] | undefined;
+  addLayer: (layer: Layer) => void;
+  removeLayer: (name: string) => void;
+  getLayerByName: (name: string) => BaseLayer | undefined;
+  aoi: Extent | null;
+  setAoi: Dispatch<SetStateAction<Extent | null>>;
+  selectedLayer: Layer | null;
+  setSelectedLayer: Dispatch<SetStateAction<Layer | null>>;
+  collections: Collection[];
+  setCollections: Dispatch<SetStateAction<Collection[] | null>>;
+};
+
+type MapProviderProps = {
+  initialState?: Partial<MapContextType>;
+  children: ReactNode;
+};
+
+export const MapContext = createContext<MapContextType | null>(null);
+MapContext.displayName = 'MapContext';
+
+const DEFAULT_LON_LAT = [-0.09, 51.505];
+const DEFAULT_ZOOM = 7;
+
+export const MapProvider = ({ initialState = {}, children }: MapProviderProps) => {
+  const mapRef = useRef(null);
+
+  const [mapConfig, setMapConfig] = useState<MapConfig>({
+    center: DEFAULT_LON_LAT,
+    zoom: DEFAULT_ZOOM,
+  });
+  const [map, setMap] = useState<Map | undefined>();
+  const [aoi, setAoi] = useState<Extent | null>(null);
+  const [collections, setCollections] = useState<Collection[] | null>(null);
+  const [selectedLayer, setSelectedLayer] = useState<Layer | null>(null);
+
+  const {
+    state: { activeFilters },
+  } = useFilters();
+
+  useDebounce(
```

Comment:
is `useDebounce` really needed here?

Reply:
I think so yes as otherwise the function is executed on every character typed. This is where you had it initially, so I tried to keep the changes limited to just swapping out Leaflet for OpenLayers.

The debouncing code could possibly move to the UI code when the user types, but that has nothing to do with OpenLayers or Leaflet, so I'd leave such a change to a separate ticket.